script.js

```diff
@@ -1,4 +1,174 @@
+// Global veriables
+var userTally = 0;
+var computerTally = 0;
+var rounds = 0;
+var userName = '';
+var gameModeChoice = '';
+var gameModeText = '';
+var gameModeMessage = `Please select a game mode: <br>
+    Enter "1" for regular Scissors Paper Stone <br>
+    Enter "2" for reversed Scissors Paper Stone <br>
+    Enter "3" for Korean Scissors Paper Stone <br>
+    Enter "4" for computer vs computer Scissors Paper Stone`;
+var mostRecentWinner = '';
+
+// Main function
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // Prompt user name
+  if (userName == '') {
+    if (input == '') {
+      return `Please input your name!`;
+    } else {
+      userName = input;
+      return `Welcome ${input}! <br>` + gameModeMessage;
+    }
+  } else {
+    // Prompt game mode
+    if (gameModeChoice == '') {
+      if (Number.isNaN(Number(input)) || input > 4 || input < 1) {
+        return `Invalid input! <br>` + gameModeMessage;
+      } else {
```

Comment:
there's no need for an else statement here as you are already returning in your if statement. This means that the code after your if statement would already only run if your if statement condition is not fulfilled. By removing the else statement, it makes your code less nested and more readable